staging/kube-oidc-proxy/templates/deployment.yaml

```diff
@@ -166,6 +172,11 @@ spec:
             items:
             - key: ca.crt
               path: oidc-ca.pem
+        {{- else if .Values.oidc.caCertPemHostPath }}
+        - name: ca-cert-pem
+          hostPath:
```

Comment:
I don't think we should rely on host path for this change. The docker image should already have openssl root ca certs installed (i checked `quay.io/jetstack/kube-oidc-proxy:v0.2.0`).

Reply:
yes it has, but you need to keep the local ca certs updated in the image then also, which means you need to update the image somewhen if you see that your pod starts to fail and the CA is the reason.
It is a beneficial thing to mount especially something from the host path as read-only, like the CA certs. As the system already should take care of regular updating the CA certs bundle.

Reply:
Sorry @sebbrandt87 but I still insist we should not use hostpath at all. The host OS might be different, and the location for those root CA certs bundle might be different. See https://golang.org/src/crypto/x509/root_unix.go#16.

The is the benefit of containers that OS dependencies are decoupled from the underlying host OS. The certs update can be done by running `update-ca-certificates` or equivalent command in the container before running the actual command. Or the container just need to be updated regularly to pick up the changes to the root CA certs.

Reply:
the root CAs PEM location can be different yes, I thought about that earlier already and I know that ;) 
But as we use here the upstream container image from kube-oidc-proxy and we have a fixed version right now, which is using alpine as distribution we know where the CAs cert is placed and we can expect that it will not change within that tag and maybe even not in the next tags. I will change the PR slightly just in case that you would like to change the mountpoint in the running container ;)

> The certs update can be done by running update-ca-certificates or equivalent command in the container before running the actual command. 
This is an anti-pattern to having consistent container images that you want to run and also adds "some" computing / operation time.

> Or the container just need to be updated regularly to pick up the changes to the root CA certs.
Yes, that is true but do we provide this for the used container image and would we tag the current container image then with the same tag? I would say no.

Reply:
@sebbrandt87 the problem is not the container image (alpine), the problem is the underlying OS (currently, we support debian, ubunto and centos/rhel), and they have different locations for saving root CA bundles. if we go with hostpath route, we'd probably have to have different configurations for the addon for different underlying OS distro, which is unacceptable to me as this is very hard to maintain.

Reply:
Also, the upstream build pipeline already update CA bundle on every release https://github.com/jetstack/kube-oidc-proxy/blob/master/Dockerfile#L5

So we just need to constantly picking up their new releases during base tech upgrade.

Reply:
@sebbrandt87 I think we should find a trade-off and to me the "update-ca-certificates" seems to be the most harmless option, although I understand its disadvantages, but there is no good solution it seems. Let's not get stuck on this and move forward as quickly as possible. 
also Jie just pointed out in slack - https://github.com/jetstack/kube-oidc-proxy/blob/master/Dockerfile#L5

Reply:
OK, if SRE really prefers this way, we could add this option.